code/game/machinery/camera/camera.dm

```diff
@@ -56,6 +60,13 @@
 	if(is_station_level(z) && prob(3) && !start_active)
 		toggle_cam(null, FALSE)
 		wires.cut_all()
+	for(var/obj/I in assembly.upgrades)
+		if(I.type == /obj/item/assembly/prox_sensor)
+			upgradeMotion()
+		if(I.type == /obj/item/stack/sheet/mineral/plasma)
+			upgradeEmpProof()
+		if(I.type == /obj/item/analyzer)
+			upgradeXRay()
```

Comment:
Это можно сделать намного красивее, например, создав айтемам прок апгрейда камеры и уже внутри него прописав для этих трёх предметов их функции. А тут вызывать просто по такому типу:
```suggestion
	for(var/obj/item/I as anything in assembly.upgrades)
		I.upgrade_camera(src)
```

Reply:
`upgrade_camera()` - просто условное название для прока, собственно, апгрейда камеры. Малф к этому ничего общего не имеет.

Reply:
Не, там просто малф использует проки upgradeEmpProof() и upgradeXRay() в своих улучшениях.

Reply:
То есть он не пересоздаёт камеры. А уже к существующим добавляет новые детали.
![изображение](https://github.com/ss220-space/Paradise/assets/39864130/4929cfa6-c1aa-406b-8270-1ed0ab139e3f)

Reply:
Хотя, скорее всего можно будет переписать код на
`	for(var/V in GLOB.cameranet.cameras)
		var/obj/machinery/camera/C = V
		if(C.assembly)
			var/upgraded = FALSE

			if(!C.isXRay())
				C.assembly.upgrades.Add(new /obj/item/analyzer(C.assembly))
				for(/obj/item/analyzer/I in C.assembled.upgrades)
					I.camera_upgrade()
				//Update what it can see.
				upgraded = TRUE

			if(!C.isEmpProof())
				C.assembly.upgrades.Add(new new /obj/item/stack/sheet/mineral/plasma(C.assembly))
				for(/obj/item/stack/sheet/mineral/plasma/I in C.assembled.upgrades)
					I.camera_upgrade()
				upgraded = TRUE

			if(upgraded)
				upgraded_cameras++`

Reply:
![image](https://github.com/ss220-space/Paradise/assets/115735095/b114d311-aa7a-439c-82e2-2f3bee3ae670)
Как найдёшь приемлимое решение, либо забьёшь - позвать можно вот тут.

Reply:
То есть proc в которм единственное действие - это return

Reply:
Я бы так и сделал. И напоминаю о `/datum/AI_Module/large/upgrade_cameras` и `/obj/machinery/camera/attackby()`.